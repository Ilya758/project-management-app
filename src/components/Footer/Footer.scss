.footer {
  height: 7vh;
  flex: 0 0 auto;
  padding: 0 10px;
  background: #262637;
  font-size: 16px;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell,
    Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #61dafb;
  box-shadow: 4px -2px 8px 0px rgba(34, 60, 80, 0.2);

  & .container-footer {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 0 auto;
    height: 100%;

    & .container-develop-users {
      display: flex;
      justify-content: space-between;
      width: 320px;

      & .users-link {
        cursor: pointer;
        text-decoration: none;
        color: #61dafb;

        & .fab.fa-github {
          margin-right: 5px;
        }
      }

      & .rsschool-link {
        height: 40px;
        width: 65px;
        background-image: url('../../assets/svg/logoSchool.svg');
        background-repeat: no-repeat;
        background-size: 100% 30px;
        background-position: left center;
      }
    }
  }
}
