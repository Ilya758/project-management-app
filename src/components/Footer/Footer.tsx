import { DEVELOPERS } from '../../constants/developers';
import './Footer.scss';

export const Footer = () => {
  const nickTemplate = () => {
    return DEVELOPERS.map((elem) => {
      return (
        <a href={elem.host} key={elem.name} className="users-link">
          <i className="fab fa-github"></i>
          {elem.name}
        </a>
      );
    });
  };

  return (
    <footer className="footer">
      <div className="container-footer">
        <span>Â© 2022</span>
        <div className="container-develop-users">{nickTemplate()}</div>
        <a
          href="https://rs.school/react/"
          target="_blank"
          className="rsschool-link"
          rel="noreferrer"
        ></a>
      </div>
    </footer>
  );
};
