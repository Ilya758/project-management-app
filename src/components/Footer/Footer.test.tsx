import { render } from '@testing-library/react';
import { Footer } from './Footer';

describe('Footer Component', () => {
  it('Should render', () => {
    const component = render(<Footer />);
    expect(component.baseElement).toMatchSnapshot();
  });
});
