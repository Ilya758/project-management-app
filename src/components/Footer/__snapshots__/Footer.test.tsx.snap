// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Should render 1`] = `
<body>
  <div>
    <footer>
      <div
        class="containerFooter"
      >
        <span>
          Â© 2022
        </span>
        <div
          class="containerDevelopUsers"
        >
          <a
            href="https://github.com/ilya758"
          >
            <i
              class="fab fa-github"
            />
            ilya758
          </a>
          <a
            href="https://github.com/inord777dev"
          >
            <i
              class="fab fa-github"
            />
            inord777dev
          </a>
          <a
            href="https://github.com/tyzikova"
          >
            <i
              class="fab fa-github"
            />
            tyzikova
          </a>
        </div>
        <a
          class="rsschool-link"
          href="https://rs.school/react/"
          rel="noreferrer"
          target="_blank"
        />
      </div>
    </footer>
  </div>
</body>
`;
